### ISSUE TITLE
stopping and continuing train.py adding new data to a trained bot

### ISSUE THREAD

**Issue by WillPhelps1** [daniel-kukiela/nmt-chatbot#41]
i was wondering if its possible to stop the training process and run it again while retaining all the prior data. Im just scared that if i stop the train process itll restart from the beginning with no steps. im also wondering if there is a way to use multiple reddit years in the making of the test.to and the test.from

**Comment by daniel-kukiela** [comment]
You can stop training process.
If you start it again, it'll begin at previous full 5k steps, so it;s best to stop right after next full 5k steps (you'll notice that script is inferencing against dev and test sets and saving hparams, stop right after that moment, i don't have a screenshot now to show you that).

You can use multiple files, just iterate list of files on top of opening a single file (https://pythonprogramming.net/building-database-chatbot-deep-learning-python-tensorflow/ - add a loop above `with open...`)

**Comment by daniel-kukiela** [comment]
You can't change data during training process. So if you change anything in training set, remove `model` folder and start training again.

**Comment by xchgaur** [comment]
but if I Have to decrease the learning rate, I can do that while training right? 
are there some defined steps to stop and start the training or I can do ctrl+c and run "python train.py" again!

**Comment by spiderwisp** [comment]
Hm, so there's no way then to take an existing model and "teach it new things"?

I guess I need to understand tensorflow more to understand this part.

**Comment by spiderwisp** [comment]
What I'm still a bit confused on is the Learning Rate. It only seems to be effective if I set it before running `prepare_data.py`. I'm not sure how to adjust this parameter once training commences.

**Comment by daniel-kukiela** [comment]
@xchgaur yes, you can change learning rate, but i highly suggest doing that in hparams file in model folder. You can change it in config and enable hparams override, but that might actually reset progress counter.
You should stop training only every 5k steps (or at the end of epochs). You know that moment bc there are `tests` and `dev` files evaluated - stop right after those evaluations.

**Comment by daniel-kukiela** [comment]
@spiderwisp Yes, you can't add data to that model, as that will change vocab file, which cannot change - word embedding won't match (network learns not vocab elements itself, but more like vocab element number).
About learning rate - response above :)

**Comment by daniel-kukiela** [comment]
You probably shouldn't touch learning rate until full epoch, than lower LR for the next one :)

**Comment by daniel-kukiela** [comment]
Nice idea!
There is a Discord server we are active in: https://discord.gg/57xEY2
